lib/src/lib.rs

```diff
@@ -389,15 +387,16 @@ impl<'a> ChunkAccessor for DefaultChunkAccessor<'a> {
             data
         };
 
-        let mut sha256 = sha2::Sha256::new();
+        let mut sha256 = Sha256::default();
         sha256.input(&data);
-        let mut sha256_digest = vec![0u8; DIGEST_SIZE];
-        sha256.result(&mut sha256_digest);
-        if sha256_digest != digest {
+        let mut vec_result = vec![0u8; DIGEST_SIZE];
+        vec_result.copy_from_slice(&sha256.result());
```

Comment:
I am not sure I've understood your question. Do you mean `sha2` API? `sha256.result()` returns `GenericArray<u8, U32>` which is essentially a generic equivalent of `[u8; 32]`, i.e. owned stack allocated array.

Your API uses `Vec<u8>` as a return type, and because I don't know how far change of vector to array will cascade, I've limited changes by utilizing copy into vector. I think it's a good chance to get rid of unnecessary allocations, but it's your decision to make.